### PR TITLE
[RLlib] Fix `use_lstm` flag for ModelV2 (w/o ModelV1 wrapping) and add it for PyTorch.

### PR DESCRIPTION
**Pull request by sven1977**
The `use_lstm` flag is currently broken for tf and was never supported for PyTorch. This PR fixes this issue by providing a ModelV1-free LSTM wrapper around any default model (tf or torch).
- Add two short learning tests for PPO w/ use_lstm=True.

Closes issue #8615 

Closes issue #8615 

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/agents/trainer.py [hunk 1]
```diff
@@ -535,8 +536,14 @@ def _setup(self, config):
         env = self._env_id
         if env:
             config["env"] = env
+            # An already registered env.
             if _global_registry.contains(ENV_CREATOR, env):
                 self.env_creator = _global_registry.get(ENV_CREATOR, env)
+            # A class specifier.
+            elif "." in env:
```
Comment (ericl) [inline rllib/agents/trainer.py:543]: This is kind of cool but can you please split it out into a separate PR?

File: rllib/models/torch/fcnet.py [hunk 2]
```diff
@@ -13,36 +13,32 @@
 logger = logging.getLogger(__name__)
 
 
-class FullyConnectedNetwork(TorchModelV2, nn.Module):
+class FullyConnectedNetwork(TorchModelV2):
```
Comment (ericl) [inline rllib/models/torch/fcnet.py:16]: Can you make sure to update the documentation to not use nn.Module either?

**Review by ericl — CHANGES_REQUESTED** [review]
comments inline

File: rllib/agents/trainer.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/trainer.py:543]: done

File: rllib/models/torch/fcnet.py (hunk 2 above)
Comment (sven1977) [reply rllib/models/torch/fcnet.py:16]: sure ... +1

File: rllib/agents/impala/vtrace_tf_policy.py [hunk 3]
```diff
@@ -177,7 +177,7 @@ def make_time_major(*args, **kw):
     values = model.value_function()
 
     if policy.is_recurrent():
-        max_seq_len = tf.reduce_max(train_batch["seq_lens"]) - 1
+        max_seq_len = tf.reduce_max(train_batch["seq_lens"])
```
Comment (sven1977) [inline rllib/agents/impala/vtrace_tf_policy.py:180]: This is actually a bug in IMPALA tf and torch.

File: rllib/models/catalog.py [hunk 4]
```diff
@@ -374,10 +379,18 @@ def track_var_creation(next_creator, **kw):
 
         if framework in ["tf", "tfe"]:
             v2_class = None
-            # try to get a default v2 model
+            # Try to get a default v2 model.
             if not model_config.get("custom_model"):
                 v2_class = default_model or ModelCatalog._get_v2_model_class(
                     obs_space, model_config, framework=framework)
+
+            if model_config.get("use_lstm"):
```
Comment (sven1977) [inline rllib/models/catalog.py:387]: Wrap with new LSTMWrappers (tf and torch).

File: rllib/models/torch/fcnet.py (hunk 2 above)
Comment (sven1977) [reply rllib/models/torch/fcnet.py:16]: done

File: rllib/models/torch/misc.py [hunk 5]
```diff
@@ -95,11 +95,11 @@ def __init__(self,
                  bias_init=0.0):
         super(SlimFC, self).__init__()
         layers = []
-        linear = nn.Linear(in_size, out_size)
+        self.linear = nn.Linear(in_size, out_size)
```
Comment (sven1977) [inline rllib/models/torch/misc.py:98]: Need to access this in LSTMs to be able to properly create the internal state tensors (on same device as the entire Model).

**Review by ericl — APPROVED** [review]
